stix2/base.py

```diff
@@ -99,3 +101,33 @@ def __repr__(self):
         props = [(k, self[k]) for k in sorted(self._properties) if self.get(k)]
         return "{0}({1})".format(self.__class__.__name__,
                                  ", ".join(["{0!s}={1!r}".format(k, v) for k, v in props]))
+
+#  Versioning API
+
+    def new_version(self, **kwargs):
```

Comment:
This is tricky. If a unsets a property that cannot be changed, the function should (and probably already does) raise an Exception. If a user unsets a property that is required with no default implementation (like `Indicator.pattern`), I'm thinking we should let the object initializer raise that error. If a user tries to unset a property that is "required", but that we have a default implementation for (like `Indicator.valid_from`), it might be confusing, since "unset" is acting more like "reset". I'm not sure what the right thing to do is.